src/MCStack.cpp

```diff
@@ -228,15 +306,55 @@ MCStack::getFirstEnabledTransition()
     // their state members on
     const bool transitionIsInSleepSet = sTop.threadIsInSleepSet(i);
     if (transitionIsEnabled && !transitionIsInSleepSet)
+      // FIXME:  We are returning an address from this local call frame.
```

Comment:
Since there may not be an enabled transition, a nullable `const MCTransition *` is desired. Here this is safe (albeit confusing) since the local is actually a reference to a transition which outlives the function scope.

That said, it's probably better to return a pointer from `getNextTransitionForThread(tid_t)` as you suggest since it is a little confusing here.

Reply:
I see, about the reference variable actually referencing something outside this scope.  But I agree that this is confusing.

If we're going to return a pointer from `getNextTransitionForThread(tid_t)`, then I'd like to simply return the type `const MCTransition *`, instead of `const std::shared_ptr<MCTransition>`.  If I understand the code, then there is only one thread that will manipulate things of type MCTransition.  (This is the single thread contained in the model checker process.)

Right now, McMini uses  `const std::shared_ptr<MCTransition>`.  This makes things difficult on two counts:
 1.  It is slower, since it must call an internal C++ method, `get()`, to dereference the pointer.
 2.  It makes life more difficult in GDB.  Every time I see a `get()` method in GDB, I have to do `ptype *OBJECT` to tell GDB to dereference the object and show me what it _really_ is.

Do you agree if we change  `const std::shared_ptr<MCTransition>` to  `const MCTransition *` everywhere (as a new commit)?